File: app/src/main/java/com/futabooo/android/booklife/screen/login/LoginPresenterImpl.java
package com.futabooo.android.booklife.screen.login;

public class LoginPresenterImpl implements LoginPresenter {
  @Override public void autoLogin() {

  }

  @Override public boolean isEmailValid(String email) {
    return email.contains("@");
  }

  @Override public boolean isPasswordValid(String password) {
    return password.length() > 0;
  }
}

File: app/src/main/kotlin/com/futabooo/android/booklife/screen/login/LoginPresenterImpl.kt
package com.futabooo.android.booklife.screen.login

class LoginPresenterImpl : LoginPresenter {
  override fun autoLogin() {

  }

  override fun isEmailValid(email: String): Boolean = email.contains("@")

  override fun isPasswordValid(password: String): Boolean = password.isNotEmpty()
}
